src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala

```diff
@@ -65,62 +68,70 @@ class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with Flat
     clusterResourcesConfig.initActionsScript.value,
     clusterFilesConfig.jupyterServerCrt.getName,
     clusterFilesConfig.jupyterServerKey.getName,
     clusterFilesConfig.jupyterRootCaPem.getName,
     clusterResourcesConfig.jupyterProxySiteConf.value,
     clusterResourcesConfig.jupyterCustomJs.value,
-    clusterResourcesConfig.jupyterGoogleSignInJs.value
-  ) ++ (
-    notebookServiceAccount(project).map(_ => List(ClusterInitValues.serviceAccountCredentialsFilename)).getOrElse(List.empty)
-  ) map(name => GcsObjectName(name))
-
-  // TODO Add cases for v2 of the cluster creation API
-  "LeonardoService" should "create a single node cluster with default machine configs" in isolatedDbTest {
-    // create the cluster
-    val clusterCreateResponse = leo.createCluster(userInfo, project, name1, testClusterRequest).futureValue
-
-    // check the create response has the correct info
-    clusterCreateResponse.serviceAccountInfo.clusterServiceAccount shouldEqual clusterServiceAccount(project)
-    clusterCreateResponse.serviceAccountInfo.notebookServiceAccount shouldEqual notebookServiceAccount(project)
-
-    // check the cluster has the correct machine configs
-    clusterCreateResponse.machineConfig shouldEqual singleNodeDefaultMachineConfig
+    clusterResourcesConfig.jupyterGoogleSignInJs.value)
 
-    // check the firewall rule was created for the project
-    computeDAO.firewallRules should contain key (project)
-    computeDAO.firewallRules(project).name.value shouldBe dataprocConfig.firewallRuleName
-
-    // should have created init and staging buckets
-    val initBucketOpt = storageDAO.buckets.keys.find(_.value.startsWith("leoinit-"+name1.value))
-    initBucketOpt shouldBe 'defined
+  lazy val initFiles = (configFiles ++ serviceAccountCredentialFile).map(GcsObjectName(_))
 
-    val stagingBucketOpt = storageDAO.buckets.keys.find(_.value.startsWith("leostaging-"+name1.value))
-    stagingBucketOpt shouldBe 'defined
-
-    // check the init files were added to the init bucket
-    val initBucketObjects = storageDAO.buckets(initBucketOpt.get).map(_._1).map(objName => GcsPath(initBucketOpt.get, objName))
-    initFiles.foreach(initFile => initBucketObjects should contain (GcsPath(initBucketOpt.get, initFile)))
-    initBucketObjects should contain theSameElementsAs (initFiles.map(GcsPath(initBucketOpt.get, _)))
-
-    // a service account key should only have been created if using a notebook service account
-    if (notebookServiceAccount(project).isDefined) {
-      iamDAO.serviceAccountKeys should contain key(samClient.serviceAccount)
-    } else {
-      iamDAO.serviceAccountKeys should not contain key(samClient.serviceAccount)
-    }
-
-    val dbInitBucketOpt = dbFutureValue { dataAccess =>
-      dataAccess.clusterQuery.getInitBucket(project, name1)
+  "LeonardoService" should "create a single node cluster with default machine configs" in isolatedDbTest {
+    forallClusterCreationMethods { (creationMethod, clusterName) =>
```

Comment:
Same deal as above: `forallClusterCreationMethods` helped with preventing code duplication and keeping the test logic more or less the same...for most cases but not all.